modules/boltlib/spec/functions/run_task_spec.rb

```diff
@@ -0,0 +1,183 @@
+          executor.expects(:run_task).with([host], executable, 'environment', 'message' => message)
+                  .returns(host: result)
+          Puppet::Pops::Types::ExecutionResult.expects(:from_bolt).with(host: result).returns(exec_result)
+
+          with_task('Test::Meta') do |task_type|
+            is_expected.to run.with_params(task_type, hostname, 'message' => message).and_return(exec_result)
+          end
+        end
+
+        it 'without args hash (for a task where this is allowed)' do
+          executor.expects(:from_uris).with(hosts).returns([host])
+          executor.expects(:run_task).with([host], anything, 'both', {}).returns(host: result)
+          Puppet::Pops::Types::ExecutionResult.expects(:from_bolt).with(host: result).returns(exec_result)
+
+          with_task('Test::Yes') do |task_type|
+            is_expected.to run.with_params(task_type, hostname).and_return(exec_result)
+          end
+        end
+
+        it 'without nodes - does not invoke bolt' do
+          executor.expects(:from_uris).never
+          executor.expects(:run_task).never
+
+          with_task('Test::Yes') do |task_type|
+            is_expected.to run.with_params(task_type, []).and_return(Puppet::Pops::Types::ExecutionResult::EMPTY_RESULT)
+          end
+        end
+      end
+
+      context 'a task name' do
+        it 'and args hash' do
+          executable = File.join(tasks_root, 'meta.sh')
+
+          executor.expects(:from_uris).with(hosts).returns([host])
+          executor.expects(:run_task).with([host], executable, 'environment', 'message' => message)
+                  .returns(host: result)
+          Puppet::Pops::Types::ExecutionResult.expects(:from_bolt).with(host: result).returns(exec_result)
+
+          is_expected.to run.with_params('test::meta', hostname, 'message' => message).and_return(exec_result)
+        end
+
+        it 'without args hash (for a task where this is allowed)' do
+          executable = File.join(tasks_root, 'yes.sh')
+
+          executor.expects(:from_uris).with(hosts).returns([host])
+          executor.expects(:run_task).with([host], executable, 'both', {}).returns(host: result)
+          Puppet::Pops::Types::ExecutionResult.expects(:from_bolt).with(host: result).returns(exec_result)
+
+          is_expected.to run.with_params('test::yes', hostname).and_return(exec_result)
+        end
+
+        it 'without nodes - does not invoke bolt' do
+          executor.expects(:from_uris).never
+          executor.expects(:run_task).never
+
+          is_expected.to run.with_params('test::yes', []).and_return(Puppet::Pops::Types::ExecutionResult::EMPTY_RESULT)
+        end
+
+        it 'with non existing task - reports an unknown task error' do
+          is_expected.to run.with_params('test::nonesuch', []).and_raise_error(/Task not found: test::nonesuch/)
+        end
+
+        context 'on a module that contains manifests/init.pp' do
+          it 'the call does not load init.pp' do
```

Comment:
@adreyer the init.pp file is aimed at this test, to ensure we don't try to load manifest code. I'll add a comment in the dummy file.

Reply:
cool